Utilize graphql's provided validations (#3)
stefanpenner/eslint-ast: projects/eslint-plugin-graphql/README.md

    {
      files: '**/*.graphql',
      parser: '@eslint-ast/eslint-plugin-graphql/parser',
      parserOptions: {
        schema: `${__dirname}/path/to/schema.graphql`
      },
      plugins: [
        '@eslint-ast/eslint-plugin-graphql',
      ],
      extends: require.resolve('@eslint-ast/eslint-plugin-graphql/recommended.js'),
    },
  ],
};
```

#### Full support for eslint-disable comments

```
# eslint-disable <rule-name>, <rule-2-name>, ... <rule-n-name>
# eslint-enable <rule-name>, <rule-2-name>, ... <rule-n-name>
# eslint-disable-line
# eslint-disable-line <rule-name>, <rule-2-name>, ... <rule-n-name>
# eslint-disable-next-line
# eslint-disable-next-line <rule-name>, <rule-2-name>, ... <rule-n-name>
```

> For our built-in-rules, they will have the name structure:

Reviewer: ```suggestion
For our built-in rules, they will have the name structure:
```

```
# eslint-disable @eslint-ast/graphql/single-top-level-query
```

For rules we run from graphql's own library they will have the following naming convention

```
# eslint-disable @eslint-ast/graphql/ExecutableDefinitionsRule
```

## vim-coc users

1. open `:CocConfig`
2. add `'graphql'` to `'eslint.filetypes'`

## List Of Validations:

> This project projects some of it's own rules:

Reviewer: ```suggestion
This project includes some of its own rules:
```

* single-top-level-query

> This project supports and executes all graphql's own validations as of version 15.3.0, they are:

Reviewer: ```suggestion
This project supports and executes all of graphql's own validations as of version 15.3.0; they are:
```
